ComponentKit/Core/CKComponentMemoizer.mm

```diff
@@ -14,11 +14,41 @@
 #import "CKComponentSubclass.h"
 #import "CKMacros.h"
 #import "CKInternalHelpers.h"
+#import "CKThreadLocalComponentScope.h"
+#import "CKComponentScopeHandle.h"
 
 #include <map>
 
 static NSString *CKComponentMemoizerThreadKey = @"CKComponentMemoizer";
 
+static bool CKCanUseMemoizedComponent(CKComponent *c, id scopeFrameIdentifier) {
```

Comment:
Instead, use `c.scopeHandle.rootIdentifier` which is a unique identifier for its scope frame that persists across state updates.